### PR TITLE
Jhalackova smo 423 zram

### PR DESCRIPTION
**Pull request by lvicoun**
### Description

Describe the overall goals of this pull request.

### Are there any relevant issues/feature requests?

* bsc#...
* jsc#SLE-...

### Is this (based on) existing content?

* old xml:id...
* old URL...

### CODE REVIEW

**Comment by lvicoun** [comment]
[Micro-zram_en.pdf](https://github.com/user-attachments/files/17293562/Micro-zram_en.pdf)

File: references/zram-configuration.xml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      <term>zram-size</term>
+      <listitem>
+        <para>
+Specify how big the zram device is compared to the host RAM size. The default value is
+<literal>zram-size = min(ram / 10, <replaceable>RAM_HARD_LIMIT</replaceable>)</literal>. It is a minimun of either where <literal>ram</literal> is the host RAM
+size obtained using <command>MemTotal</command>. 
+        </para>
+        <itemizedlist>
+          <listitem>
+            <para><literal>ram</literal> &ndash; is the total amount of the host memory in MB</para>
```
Comment (dariavladykina) [inline references/zram-configuration.xml:69]:
```suggestion
            <para><literal>ram</literal> is the total amount of the host memory in MB.</para>
```

File: references/zram-configuration.xml [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+          </listitem>
+          <listitem>
+            <para>
+              <literal>ram/10</literal> &ndash; is the ratio of total host memory that can be used
+              for zram. For example, the example can use 10% of the total host memory. We recommend
+              fractions of the host RAM in the range 10 &ndash; 50%.
+            </para>
```
Comment (dariavladykina) [inline references/zram-configuration.xml:76]:
```suggestion
              <literal>ram/10</literal> is the ratio of total host memory that can be used
              for zram. For example, the example can use 10% of the total host memory. We recommend
              fractions of the host RAM in the range of 10&ndash;50%.
            </para>
```

File: references/zram-configuration.xml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+          </listitem>
+          <listitem>
+            <para>
+              <replaceable>RAM_HARD_LIMIT</replaceable> &ndash; is the maximum amount of RAM in MB
+              that can be used by zram.
```
Comment (dariavladykina) [inline references/zram-configuration.xml:81]:
```suggestion
              <replaceable>RAM_HARD_LIMIT</replaceable> is the maximum amount of RAM in MB
              that can be used by zram.
```

File: references/zram-configuration.xml [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+            </para>
+          </listitem>
+          <listitem>
+            <para><literal>min()</literal> &ndash; is a function that compares the value of
+            <replaceable>RAM_HARD_LIMIT</replaceable> and the calculated ratio and select the lower
+            one. </para>
```
Comment (dariavladykina) [inline references/zram-configuration.xml:87]:
```suggestion
            <para><literal>min()</literal> is a function that compares the value of
            <replaceable>RAM_HARD_LIMIT</replaceable> and the calculated ratio 
            and selects the lower one. </para>
```

File: references/zram-configuration.xml [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+          </listitem>
+        </itemizedlist>       
+      </listitem>
+    </varlistentry>
+    <varlistentry>
+      <term>compression-algorithm</term>
+      <listitem>
+        <para>
+          Specify the alogirthm used to compress data in zram. You can use any of the following values:
```
Comment (dariavladykina) [inline references/zram-configuration.xml:96]:
```suggestion
          Specify the algorithm used to compress data in zram. You can use any of the following values:
```

File: references/zram-configuration.xml [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        </para>
+        <itemizedlist>
+          <listitem>
+            <para>
+              <literal>deflate</literal>&mdash;a lossless data compression file format
```
Comment (dariavladykina) [inline references/zram-configuration.xml:101]:
```suggestion
              <literal>deflate</literal> &mdash; a lossless data compression file format
```

File: references/zram-configuration.xml [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+            </para>
+          </listitem>
+          <listitem>
+            <para>
+              <literal>lz4</literal> or <literal>lz4hc</literal>&mdash;a lossless data compression algorithm that is focused on compression and decompression speed
```
Comment (dariavladykina) [inline references/zram-configuration.xml:106]:
```suggestion
              <literal>lz4</literal> or <literal>lz4hc</literal> &mdash; a lossless data compression algorithm that is focused on compression and decompression speed
```

File: references/zram-configuration.xml [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+            </para>
+          </listitem>          
+          <listitem>
+            <para>
+              <literal>lzo-rle</literal>&mdash;a lossless data compression algorithm that is focused on decompression speed
```
Comment (dariavladykina) [inline references/zram-configuration.xml:111]:
```suggestion
              <literal>lzo-rle</literal> &mdash; a lossless data compression algorithm that is focused on decompression speed
```

File: references/zram-configuration.xml [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+            </para>
+          </listitem>
+          <listitem>
+            <para>
+              <literal>zstd</literal> aka Zstandard&mdash;a lossless data compression algorithm
```
Comment (dariavladykina) [inline references/zram-configuration.xml:116]:
```suggestion
              <literal>zstd</literal> aka Zstandard &mdash; a lossless data compression algorithm
```

File: references/zram-configuration.xml [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+              comparable to DEFLATE with the compression ratio but faster during decompression
+              </para>
+          </listitem>
+          <listitem>
+            <para>
+              <literal>842</literal>&mdash;a data compression algorithm
```
Comment (dariavladykina) [inline references/zram-configuration.xml:122]:
```suggestion
              <literal>842</literal> &mdash; a data compression algorithm
```

Comment (dariavladykina) [reply references/zram-configuration.xml:122]: This also goes for the list in `section xml:id="zram-about-benefits`. It is visually better to use spaces around the em dash, and I will update the style guide accordingly.

File: references/zram-configuration.xml [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+        <para>
+         Specify a device where to store incompressible pages. These pages are then removed from zram. 
+        </para>
+      </listitem>
+    </varlistentry>
+    <varlistentry>
+      <term>fs-type</term>
+      <listitem>
+        <para>
+          If the device is used as a disk, specify file system to be used there. For example, <literal>ext4</literal>.
```
Comment (dariavladykina) [inline references/zram-configuration.xml:155]:
```suggestion
          If the device is used as a disk, specify the file system to be used there. For example, <literal>ext4</literal>.
```

File: references/zram-configuration.xml [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+      </listitem>
+    </varlistentry>
+  </variablelist>
+  <section xml:id="zram-configuration-main-config-example">
+    <title>An example of <filename>/etc/systemd/zram-generator.conf</filename></title>
+    <para>
+      The following example creates two block device in RAM&mdash;<literal>zram0</literal> for swap
+      and <literal>zram1</literal> with EXT4 file system. If the machine has more than 9&nbsp;GB of RAM,
+      <literal>zram0</literal> will not be created. Each of the block device will allocate up to
+      2&nbsp;GB of RAM.
```
Comment (dariavladykina) [inline references/zram-configuration.xml:175]:
```suggestion
      The following example creates two block devices in RAM &mdash; <literal>zram0</literal> for swap
      and <literal>zram1</literal> with the ext4 file system. If the machine has over 9&nbsp;GB of RAM,
      <literal>zram0</literal> will not be created. Each block device will allocate up to 2&nbsp;GB of RAM.
```

File: references/zram-configuration.xml (hunk 2 above)
Comment (dariavladykina) [reply references/zram-configuration.xml:76]: Just checking if the text _For example, the example can use [...]_ is legit :-)

File: tasks/zram-creating-systemd-unit.xml [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+    <!-- add author's e-mail -->
+    <meta name="maintainer" content="jsindelarova@suse.com" its:translate="no"/>
+    <abstract><!-- can be changed via merge in the assembly -->
+      <para>
+       The topic covers the last steps to take to create zram in RAM.
+      </para>
+    </abstract>
+  </info>
+  <para>
+    After you prepare the configuration, create and start the devices. To do, see the procedure below:
```
Comment (dariavladykina) [inline tasks/zram-creating-systemd-unit.xml:32]:
```suggestion
    After you prepare the configuration, create and start the devices. To do this, 
    see the procedure below:
```

**Pull request merged**